src/main/scala/mesosphere/marathon/core/task/tracker/impl/AppData.scala

```diff
@@ -32,9 +33,8 @@ private[impl] case class AppDataMap private (appTasks: Map[PathId, AppData]) {
     task <- app.taskMap.get(taskId)
   } yield task
 
-  def toTaskTrackerAppMap: Map[PathId, TaskTracker.App] = {
-    appTasks.mapValues(_.toTaskTrackerApp)
-  }
+  def toTaskTrackerAppMap: Map[PathId, TaskTracker.App] = appTasks.mapValues(_.toTaskTrackerApp)
+  def tasks: Iterable[MarathonTask] = appTasks.view.flatMap(_._2.tasks)
```

Comment:
If I change that now, I will get into major conflicts. I created another issue for this:

https://github.com/mesosphere/marathon/issues/3003